File: android/app/src/main/java/org/getlantern/lantern/service/AutoStarter.java
package org.getlantern.lantern.service;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;

import org.getlantern.mobilesdk.Logger;

public class AutoStarter extends BroadcastReceiver {
    private static final String TAG = AutoStarter.class.getName();

    @Override
    public void onReceive(Context context, Intent intent) {
        Logger.debug(TAG, "Automatically starting Lantern Service on: " + intent.getAction());

        Intent serviceIntent = new Intent(context, LanternService_.class);
        serviceIntent.putExtra(
                LanternService.AUTO_BOOTED,
                intent.getAction() == Intent.ACTION_BOOT_COMPLETED
        );

        context.startService(serviceIntent);
    }
}

File: android/app/src/main/kotlin/org/getlantern/lantern/service/AutoStarter.kt
package org.getlantern.lantern.service

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import org.getlantern.mobilesdk.Logger

open class AutoStarter : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        Logger.d(TAG, "Automatically starting Lantern Service on: ${intent.action}")
        val serviceIntent = Intent(context, LanternService_::class.java)
            .putExtra(LanternService.AUTO_BOOTED, intent.action == Intent.ACTION_BOOT_COMPLETED)
        context.startService(serviceIntent)
    }

    companion object {
        private val TAG = AutoStarter::class.java.simpleName
    }
}
